kong/init.lua

```diff
@@ -671,10 +671,26 @@ function Kong.init()
   db:close()
 
   require("resty.kong.var").patch_metatable()
+
+  if config.privileged_agent and is_data_plane(config) then
+    local ok, err = process.enable_privileged_agent(2048)
+    if not ok then
+      error(err)
+    end
+  end
+
+  -- because of resurrection, objects with finalizers are collected in two phases
+  collectgarbage()
+  collectgarbage()
 end
 
 
 function Kong.init_worker()
+  if process.type() == "privileged agent" then
```

Comment:
`process.type() == "privileged agent"` appears many times, could we move it into a `utils.is_privileged_agent()`?

Reply:
@chronolaw it almost start to feel that we should just make `utils` part of the `kong` global. So that we don't need to add `require "kong.tools.utils"` on each file.

Reply:
I added `TODO: ...`.